### PR TITLE
 refactor: break conversion out into separate files and use Concentrate

### PR DESCRIPTION
**Pull request by sbender9**

### CODE REVIEW

**Comment by mairas** [comment]
Would it make sense if each individual conversion module would in their `module.exports` return a list of functions instead of a single function? I'd like to create conversions for every environment temperature variable. They all map to the same PGN with different `tempSource` values. I'd like to map the (SK path, tempSource) values to separate functions, and returning a list would make that possible.

I'm a complete newbie in JS, so it may well be I've missed something trivial here...

File: index.js [hunk 1]
```diff
@@ -3,146 +3,191 @@ const debug = require("debug")("signalk:signalk-to-nmea2000");
...
 
 module.exports = function(app) {
   var plugin = {};
   var unsubscribes = [];
-  var timer;
+  var timers = []
+  var conversions = load_conversions(app, plugin)
+  conversions = [].concat.apply([], conversions)
+
+  var types = {
```
Comment (mairas) [inline index.js]: It seems like the conversion types are an essential concept when defining the conversions. I struggled to understand their differences by following the code only. Could you maybe add a some comments clarifying the different conversion types?

**Comment by sbender9** [comment]
It now supports the conversion returning an error

**Comment by sbender9** [comment]
I've also changed the type stuff and added a comment to document it.

**Comment by sbender9** [comment]
Don’t do anything with the arrays yet. I need to change the way that’s done.

**Comment by sbender9** [comment]
ok @mairas take a look at the battery conversion. Yours will look something like that.

File: index.js [hunk 2]
```diff
@@ -3,146 +3,240 @@ const debug = require("debug")("signalk:signalk-to-nmea2000");
...
+  conversions = [].concat.apply([], conversions)
+
+  /*
+    Each conversion can specify the sourceType and outputType.
+
+    Source type can be:
+
+      onDelta - You will get all deltas via app.signalk.on('delta', ...). Please do no use this unless absolutely necessary.
+      
+      bacanjs - The conversion should specify a variable called 'keys' which is an array of the Signal K paths that the convesion needs
```
Comment (mairas) [inline index.js]: Could this be renamed to e.g. `onValueChange`? As far as I understood, `baconjs` refers to specific implementation, not the source concept.

Also: typo (on line 33, too).

**Comment by mairas** [comment]
The sub-list approach looks good to me! Thanks!

File: index.js (hunk 2 above)
Comment (tkurki) [reply index.js]: +1 to `onValueChange` or similar

File: conversions/battery.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+          }
+        }
+      }
+    },
+
+    conversions: (options) => {
+      return options.BATTERY.batteries.map(battery => {
+        return {
+          keys: batteryKeys.map(key => `electrical.batteries.${battery.signalkId}.${key}`),
+          timeouts: batteryKeys.map(key => 60000),
```
Comment (tkurki) [inline conversions/battery.js]: If all the timeouts are the same maybe a single `number` would do here and the calling code could handle arrays and single numbers. Pretty minor...

**Pull request merged**